integration_tests/testsuite/deploy_app.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/python
+
+# Copyright 2016 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import subprocess
+import sys
+import time
+
+DEPLOY_DELAY_SECONDS = 30  # time to give GAE to start app after deploy
+
+
+def cleanup(appdir):
+    try:
+        os.remove(os.path.join(appdir, 'Dockerfile'))
+    except:
+        pass
+
+
+def _deploy_app(image, appdir):
+    try:
+        # change to app directory (and remember original directory)
+        owd = os.getcwd()
+        os.chdir(appdir)
```

Comment:
since I also have to read/write Dockerfiles in that directory, it's a bit easier to just move into the directory beforehand. unless you have strong opposition to doing it this way I'll leave it as is.